### ISSUE TITLE
torch.fx to MDF

### ISSUE THREAD

**Issue by davidt0x** [ModECI/MDF#431]
The current pipeline for PyTorch to MDF conversion utilizes TorchScript compilation of PyTorch models. The TorchScript graph is created with a call to `torch.jit.script`. The TorchScript graph is then traversed and converted to an MDF model in a node by node manner (for the most part). The code for this is all contained in [modeci_mdf.interfaces.pytorch.importer](https://github.com/ModECI/MDF/blob/main/src/modeci_mdf/interfaces/pytorch/importer.py). The future direction of PyTorch 2.0 [PyTorch 2.0](https://pytorch.org/get-started/pytorch-2.0/) seems to be moving away from TorchScript in favor or [torch.fx](https://pytorch.org/docs/stable/fx.html) graphs and [torch.compile](https://pytorch.org/tutorials/intermediate/torch_compile_tutorial.html#torchdynamo-and-fx-graphs). It would probably be best to rewrite the TorchScript compilation dependency to utilize these new features. This might have the added advantage that FX graph IR seems to capture some higher level details (hierarchy of `nn.Module`) of the model which might be carried over to MDF. A good first step would be to implement a simple proof of concept conversion from a simple PyTorch example model to an MDF model. A good first example would be this simple [model](https://github.com/ModECI/MDF/blob/1901625c8b360cbb468428a711965223959c4bcf/examples/PyTorch/simple_pytorch_to_mdf.py#L12). 

A proof of concept or rough implementation of this feature for this simple model would be a great thing to try for an Outreachy candidate. A a more complete implementation could be pursued by a successful candidate over the summer.
